### PR TITLE
feat/#460 EIP-4895 Validators withdraw from Beacon chain - part 1

### LINKED ISSUE #460: [Shanghai] EIP-4895 Beacon chain push withdrawals as operations
[EIP-4895: Beacon chain push withdrawals as operations](https://eips.ethereum.org/EIPS/eip-4895)

### PR DESCRIPTION
**Pull request by KimiWu123**
This PR is part of #460.

I'll separate #460 into two parts, 
1. spec verification
2. python code implementation

This PR is trying to fix the first part: spec verification

### CODE REVIEW

File: specs/withdrawal-proof.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+|                |                | *WDContextFieldTag* |               |
+| $WithdrawalID  | $ValidatorID   | Address             | $value{Lo,Hi} |
+| $WithdrawalID  | $ValidatorID   | Amount              | $value{Lo,Hi} |
+
+There are some constraints on the shape of the table like:
+
+- For every withdrawal, each tag must appear exactly once.
+- `WithdrawalID` is increased monotonically and sequentially for each withdrawal. Which means it won't be starting from 0 in most of cases.
+
+Since the withdrawal table is built from public inputs, the verifier (the L1 smart contract in the zkRollup for example) needs to validate that all rows of the table are properly built using the withdrawal data.  Since the table construction is validated outside of the circuit, there's no need to verify the same constraints inside of the circuit.
```
Comment (hero78119) [inline specs/withdrawal-proof.md:50]: `WithdrawalRoot` contains ONLY the withdraws operations of this block. So I think public input we just need to add one extra field `WithdrawalRoot`, and the withdrawal table can just be witness from prover, and rely on the validation of `mpt(withdrawalsData)` via lookup into mpt table.

Comment (hero78119) [reply specs/withdrawal-proof.md:50]: Based on this, in `super_circuit.png` public input for withdrawls can just one single 256 bit field `WithdrawalRoot` :)

File: specs/withdrawal-proof.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Withdrawals proof
```
Comment (hero78119) [inline specs/withdrawal-proof.md:1]: Looks like we also need to docs on how to handle balance update on address

**Review by hero78119 — COMMENTED** [review]
First pass and looks pretty nice to me! 🔥 🔥  really excited to see withdrawls functionality introduce!

File: specs/withdrawal-proof.md (hunk 2 above)
Comment (KimiWu123) [reply specs/withdrawal-proof.md:1]: fixed in [f4e0cfc](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/487/commits/f4e0cfc357419830f179c91cad9aebaf0da8e11a)

File: specs/withdrawal-proof.md (hunk 1 above)
Comment (KimiWu123) [reply specs/withdrawal-proof.md]: `super_circuit.png` updated in [e19a56c](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/487/commits/e19a56cf56c3060ea4e0be58d080daa1d17c5d22)

File: specs/withdrawal-proof.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+For each withdrawal, the withdrawal circuit also must prepare the key and value used to build the withdrawal trie.  These keys and values are used in lookups to the MPT table in order to verify that a tree built with the key-values corresponding to the withdrawals has the root value `withdrawalsRoot`.
+
+> By doing lookups to the MPT table, we prove that when we start with an empty MPT, and do a chain of key-value insertions corresponding to each transaction, we reach a Trie with root value `withdrawalsRoot`.
+
+Each MPT update uses the following parameters:
+
+- Key = `rlp(withdrawal_index)`
+- Value = `rlp([withdrawal_index, validator_index, address, amount])`
+- ValuePrev = `rlp([withdrawal_index_prev, validator_index_prev, address_prev, amount_prev])`
```
Comment (han0110) [inline specs/withdrawal-proof.md:61]: Since it's a stack trie, I think `ValuePrev` should always be empty (non-exist)?

**Review by han0110 — APPROVED** [review]
Overall LGTM, just a nitpick on stack trie update.

File: specs/withdrawal-proof.md (hunk 3 above)
Comment (KimiWu123) [reply specs/withdrawal-proof.md]: fixed in [3444436](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/487/commits/34444362e060d2f85525c48a8557d79a6017c836)

**Review by hero78119 — APPROVED** [review]
👍 LGTM!

**Pull request merged**